python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
+        cls,
+        config: Union[ScalingConfig, "ScalingConfigDataClass"],
```

Comment:
Another idea is to have each `Trainer` define its own `ScalingConfigDataClass` subclass, so that users can tell from the public documentation exactly what's supported.

Reply:
Yeah we talked about this approach in the past, but I think for now we decided to just have a single ScalingConfig

```
Decision:
Start with sidestep: Trainer(
	**rllib_specific_configs, 
	scaling_config=ScalingConfig(**common_configs))
Let's revisit this in the future: RLlibScalingConfig(**rllib_specific_configs, **common_configs)
```